client/state/i18n/util.js

```diff
@@ -0,0 +1,16 @@
+/**
+ * External dependencies
+ */
+import { kebabCase, upperCase } from 'lodash';
+
+/**
+ * Generates (naively) a locale in the format ll-CC.
+ * Result may or may not be valid.
+ *
+ * @param  {String} lang WP.com lang/locale code
+ * @param  {String} countryCode GeoIP country code
+ * @return {String}             Naive concat of lang + countryCode
+ */
+export const generateLangCcLocale = ( lang, countryCode ) => {
+	return kebabCase( lang ) + '-' + upperCase( countryCode );
```

Comment:
Yeah...    it's a bit of a kludge but that stems from the uncertainty of the WP.com "locale" (lang?) value that half of it is powered by.  It's intentionally naive with no normalization of the values (beyond case) as they come into it.

The dashed values -- ie, WP.com locales proper like `pt-br`, `zh-tw`, etc -- generate nonsense but that's ok with how we store and look up the data. In those cases, the correct values would get returned in the fallback lookup at the "lang" (wp.com code) level.


`pt-br` WP.com lang setting in NY:
`pt-br-US`

`de` WP.com lang setting in NY:
`de-US`

`de` WP.com lang setting in DE:
`de-DE`